### PR TITLE
[graphql/rpc] add tx data loader

### PR DESCRIPTION
**Pull request by oxade**
## Description 

* First impl of [data loader paradigm](https://github.com/graphql/dataloader) for tx gets to reduce data layer round trips
* Scalar type for tx digest

## Test Plan 

Manual

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/types/tx_digest.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+impl<T: ?Sized, H, R> SerializeAs<T> for Readable<H, R>
+where
+    H: SerializeAs<T>,
+    R: SerializeAs<T>,
+{
+    fn serialize_as<S>(value: &T, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if serializer.is_human_readable() {
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/types/tx_digest.rs:35]: Just curious, how would someone ask for a readable form in the query?

File: crates/sui-graphql-rpc/src/server/simple_server.rs [hunk 2]
```diff
@@ -17,6 +19,7 @@ use std::time::Duration;
 
 pub(crate) const RPC_TIMEOUT_ERR_SLEEP_RETRY_PERIOD: Duration = Duration::from_millis(10_000);
 pub(crate) const MAX_CONCURRENT_REQUESTS: usize = 1_000;
+pub(crate) const DATA_LOADER_LRU_CACHE_SIZE: usize = 1_000;
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/server/simple_server.rs:22]: Does this indicate that the cache holds 1000 items?

File: external-crates/move/Cargo.lock
Comment (stefan-mysten) [inline external-crates/move/Cargo.lock:1]: Is this really related to this transaction digest changes?

**Review by stefan-mysten — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/tx_digest.rs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+use std::marker::PhantomData;
+
+const SUI_TRANSACTION_DIGEST_LENGTH: usize = 32;
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Copy)]
+pub(crate) struct TransactionDigest(
+    #[serde_as(as = "Readable<Base58, Bytes>")] [u8; SUI_TRANSACTION_DIGEST_LENGTH],
+);
+scalar!(TransactionDigest, "TransactionDigest");
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/tx_digest.rs:19]: It's fine to have new helper types in the implementation, but we don't want to introduce new scalar types -- is there a way to avoid this affecting the external schema?

**Review by amnn — COMMENTED** [review]
We don't want to introduce a dedicated scalar type for `TransactionDigest` and the comment about the change to `move/Cargo.lock` seems legit -- looks like this was landed a bit early?

**Comment by amnn** [comment]
Auto-merge and accepting with comments are both great ways to avoid time blocks, but when used together, it's kind of disastrous.  Can we save using auto-merge for small fixes etc? The alternative of not accepting with comments will slow things down too much, I think.
